subsys/bluetooth/host/hci_core.c

```diff
@@ -4686,6 +4659,36 @@ int bt_addr_le_create_static(bt_addr_le_t *addr)
 	return 0;
 }
 
+static u8_t bt_read_public_addr(bt_addr_le_t *addr)
+{
+	struct bt_hci_rp_read_bd_addr *rp;
+	struct net_buf *rsp;
+	int err;
+
+	/* Read Bluetooth Address */
+	err = bt_hci_cmd_send_sync(BT_HCI_OP_READ_BD_ADDR, NULL, &rsp);
+	if (err) {
+		return 0U;
+	}
+
+	rp = (void *)rsp->data;
+
+	BT_DBG("status 0x%02x", rp->status);
+
+	if (rp->status ||
```

Comment:
`bt_hci_cmd_send_sync()` will return an error if the command complete status was non-zero, so `rp->status` will never be non-zero at this point. (it's a separate discussion whether this is a good thing, since it makes it impossible to catch the exact HCI status later, the HCI command API only has two possible mappings for it, if you check the implementation)

Reply:
Correct, thanks.

I included the Logging of the `BT_DBG("status 0x%02x", rp->status);` from bdaddr and mistankenly thought I had to handle this status.
Looking over the code there are more instances similar to `read_baddr_complete` that logs the HCI status code, but it is no point, because the return error code has to be handled and the buf is always unref. So we could remove all of those.